File: assertion-roulette/python-unittest/GlobalQueryFiltersTests_Should_Add_Creation_Auditing_Properties.py
import unittest

class TestGlobalQueryFilters(unittest.TestCase):
    def test_should_add_creation_auditing_properties(self):
        result = default_test_db_context.products.add(Product(name="Product Name", code="product_code"))
        default_test_db_context.save_changes()
        
        self.assertIsNotNone(result.entity.creator_user_id)
        self.assertNotEqual("", result.entity.creator_user_id)
        self.assertEqual(datetime.today().strftime("%Y-%m-%d"), result.entity.creation_time.strftime("%Y-%m-%d"))

File: assertion-roulette/csharp-xunit/GlobalQueryFiltersTests_Should_Add_Creation_Auditing_Properties.cs
using Xunit;

namespace BackOfficeBase.Tests.DataAccess
{
    public class GlobalQueryFiltersTests
    {
        [Fact]
        public void Should_Add_Creation_Auditing_Properties()
        {
            {
                var result = await DefaultTestDbContext.Products.AddAsync(new Product { Name = "Product Name", Code = "product_code" });
                await DefaultTestDbContext.SaveChangesAsync();
                Assert.NotNull(result.Entity.CreatorUserId);
                Assert.NotEqual(Guid.Empty, result.Entity.CreatorUserId);
                Assert.Equal(DateTime.Today.ToShortDateString(), result.Entity.CreationTime.ToShortDateString());
            }
        }
    }
}
